### PR TITLE
timeout for tpu check

### LINKED ISSUE #4324: Hanging when importing pytorch_lightning on google cloud vm.
## ❓ Questions and Help

Hi,

I was trying to use pytorch-lightning with TPU on google cloud virtual machine and the virtual machine is created by this command:
```
gcloud compute instances create tpu-vm \
	   --machine-type=n1-standard-4 \
	   --image-project=ml-images \
	   --image-family=torch-xla \
	   --boot-disk-size=200GB \
   --scopes=cloud-platform
```
 When I try to import pytorch_lightning, it hanged all the time. I tried on both jupyter notebook and python in terminal, the results are the same. When I KeyboardInterrupt the code, it returns the logs below. It seems that it has something to do with the multiprocessing part. I have tried different installation methods but still have this problem. Could you help me to solve this problem? Any ideas or help would be much appreciated! Thanks!

```
---------------------------------------------------------------------------
KeyboardInterrupt                         Traceback (most recent call last)
<ipython-input-1-702a026384b9> in <module>
      1 import torch_xla.core.xla_model as xm
----> 2 import pytorch_lightning as pl
      3 # from pytorch_lightning import Trainer, seed_everything

/anaconda3/envs/ldetr/lib/python3.6/site-packages/pytorch_lightning/__init__.py in <module>
     54     # We are not importing the rest of the lightning during the build process, as it may not be compiled yet
     55 else:
---> 56     from pytorch_lightning.core import LightningDataModule, LightningModule
     57     from pytorch_lightning.callbacks import Callback
     58     from pytorch_lightning.trainer import Trainer

/anaconda3/envs/ldetr/lib/python3.6/site-packages/pytorch_lightning/core/__init__.py in <module>
     14 
     15 from pytorch_lightning.core.datamodule import LightningDataModule
---> 16 from pytorch_lightning.core.lightning import LightningModule
     17 
     18 __all__ = [

/anaconda3/envs/ldetr/lib/python3.6/site-packages/pytorch_lightning/core/lightning.py in <module>
     44 
     45 
---> 46 TPU_AVAILABLE = XLADeviceUtils.tpu_device_exists()
     47 
     48 if TPU_AVAILABLE:

/anaconda3/envs/ldetr/lib/python3.6/site-packages/pytorch_lightning/utilities/xla_device_utils.py in tpu_device_exists()
     88         """
     89         if XLADeviceUtils.TPU_AVAILABLE is None and TORCHXLA_AVAILABLE:
---> 90             XLADeviceUtils.TPU_AVAILABLE = pl_multi_process(XLADeviceUtils._is_device_tpu)()
     91         return XLADeviceUtils.TPU_AVAILABLE

/anaconda3/envs/ldetr/lib/python3.6/site-packages/pytorch_lightning/utilities/xla_device_utils.py in wrapper(*args, **kwargs)
     41         proc = Process(target=inner_f, args=(queue, func,), kwargs=kwargs)
     42         proc.start()
---> 43         proc.join()
     44         return queue.get()
     45 

/anaconda3/envs/ldetr/lib/python3.6/multiprocessing/process.py in join(self, timeout)
    122         assert self._parent_pid == os.getpid(), 'can only join a child process'
    123         assert self._popen is not None, 'can only join a started process'
--> 124         res = self._popen.wait(timeout)
    125         if res is not None:
    126             _children.discard(self)

/anaconda3/envs/ldetr/lib/python3.6/multiprocessing/popen_fork.py in wait(self, timeout)
     48                     return None
     49             # This shouldn't block if wait() returned successfully.
---> 50             return self.poll(os.WNOHANG if timeout == 0.0 else 0)
     51         return self.returncode
     52 

/anaconda3/envs/ldetr/lib/python3.6/multiprocessing/popen_fork.py in poll(self, flag)
     26             while True:
     27                 try:
---> 28                     pid, sts = os.waitpid(self.pid, flag)
     29                 except OSError as e:
     30                     # Child process not yet created. See #1731717

KeyboardInterrupt: 
```

### PR DESCRIPTION
**Pull request by lezwon**
## What does this PR do?

Fixes #4324 
This PR adds a timeout of 10 secs to the process which is used to fetch the type of XLA device. This ensures that the process does not hang indefinitely and does not block the importing of lightning as in the case of #4324.

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.    
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/utilities/xla_device_utils.py [hunk 1]
```diff
@@ -38,10 +39,13 @@ def pl_multi_process(func):
...
-        proc = Process(target=inner_f, args=(queue, func,), kwargs=kwargs)
+        proc = Process(target=inner_f, args=(queue, func, *args), kwargs=kwargs)
         proc.start()
-        proc.join()
-        return queue.get()
+        proc.join(10)
+        try:
+            return queue.get_nowait()
+        except q.Empty:
+            return False
```
Comment (lezwon) [inline pytorch_lightning/utilities/xla_device_utils.py:48]: @awaelchli should we raise an exception here instructing the user to set up TPU's properly? i.e adding `TPU_IP_ADDRESS`

File: pytorch_lightning/utilities/xla_device_utils.py [hunk 2]
```diff
@@ -24,9 +25,9 @@
     xm = None
 
 
-def inner_f(queue, func, **kwargs):  # pragma: no cover
+def inner_f(queue, func, *args, **kwargs):  # pragma: no cover
     try:
-        queue.put(func(**kwargs))
+        queue.put(func(*args, **kwargs))
     except Exception as _e:
```
Comment (Borda) [inline pytorch_lightning/utilities/xla_device_utils.py:31]:
```suggestion
    except Exception:
```

File: tests/utilities/test_xla_device_utils.py [hunk 3]
```diff
@@ -11,13 +11,17 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import math
+import time
+
 import pytest
 
-from pytorch_lightning.utilities.xla_device_utils import XLADeviceUtils as xdu
+import pytorch_lightning.utilities.xla_device_utils as utils
```
Comment (Borda) [inline tests/utilities/test_xla_device_utils.py:19]:
```suggestion
import pytorch_lightning.utilities.xla_device_utils as xla_utils
```
to be clear what utils...

File: tests/utilities/test_xla_device_utils.py [hunk 4]
```diff
@@ -42,3 +46,14 @@ def test_xla_device_is_a_tpu():
...
     return device_type == "TPU"
+
+
+def test_result_returns_within_10_seconds():
+    """Check that pl_multi_process returns within 10 seconds"""
+
+    start = time.time()
+    result = utils.pl_multi_process(time.sleep)(25)
+    end = time.time()
+    elapsed_time = math.floor(end - start)
```
Comment (Borda) [inline tests/utilities/test_xla_device_utils.py:57]:
```suggestion
    elapsed_time = int(end - start)
```

File: tests/utilities/test_xla_device_utils.py [hunk 5]
```diff
@@ -42,3 +46,14 @@ def test_xla_device_is_a_tpu():
...
+    assert elapsed_time <= 10
```
Comment (Borda) [inline tests/utilities/test_xla_device_utils.py:58]: what is the reason after rounding down to accept 10, it means that also 10.9 is valid?

Comment (lezwon) [reply tests/utilities/test_xla_device_utils.py:58]: yep. the extra time is for the instructions before and after `.join(10)`

File: pytorch_lightning/utilities/xla_device_utils.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/utilities/xla_device_utils.py:48]: maybe, could there be other  reasons for a timeout? is there also a TPU_MASTER_PORT or similar?

**Comment by edenlightning** [comment]
@lezwon mind rebasing?

File: pytorch_lightning/utilities/xla_device_utils.py (hunk 1 above)
Comment (lezwon) [reply pytorch_lightning/utilities/xla_device_utils.py:48]: Nah. Don't see any ENV var with that name.

**Review by awaelchli — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Pull request merged**